### DISCUSSION TITLE
SVG Animations

### DISCUSSION THREAD

**Discussion by IlyaTaidi** [nandorojo/moti#173] [Q&A]
Hello, I had some questions about SVG Animations within the context of Moti & Expo 

1. I don't see SVG animations anywhere in the docs, which docs (that i assume are under moti's hood) should I look for to get the best syntax/practice?
2. Should I be using SVGR or Animate the SVG's another way?
3. Assuming I want to have complex animations, should each Animated Item be its own component? Or should the Component be a tsx object that holds all the animations for that given scene?
4. If I have multiple scenes, and I animate between them, should I animate each item individually with Moti? Or load a before and after SVG and have moti go from before to after by itself?

**Comment by nandorojo** [comment]
I don’t know much about SVG animations honestly. In most cases, I’m not totally sure where moti would fit in. My guess is you would just use reanimated directly with react-native-svg, which there are many many examples for.

**Reply by IlyaTaidi** [reply]
If we're built with on top of framer motion, can we make it such that framer motion syntax works?
I'm trying to create the line-drawing animation (and tie it to a Moti Pressable press&hold animation) but having a tough time, Cant figure out where to put pathLength, pathSpacing and pathOffset.

https://codesandbox.io/s/framer-motion-5-1-line-drawing-ph6ln?from-embed

**Reply by nandorojo** [reply]
That's very interesting. It's certainly a custom feature, though, since that would use animatedProps from reanimated.

This is certainly possible with `useMotiPressableAnimatedProps`, if an animated SVG is the child of a MotiPressable

**Reply by nandorojo** [reply]
> Cant figure out where to put pathLength, pathSpacing and pathOffset.

These would be passed to `useAnimatedProps`, which then get passed to the `animateProps` of a component.

**Reply by nandorojo** [reply]
for starters, you need to learn to use react-native-svg without any animations. once that's done, animating it should be somewhat easy I assume. Will Candillon has some nice SVG animation videos

**Reply by IlyaTaidi** [reply]
Thank you so much, I'll read those docs and watch those vids.
